### DISCUSSION TITLE
Customize internal links in Lexical rich-text

### DISCUSSION THREAD

**Discussion by unonweb** [payloadcms/payload#4151] [Q&A]
Hi community,
I'd like to customize or replace the internal link feature. At the moment my users have to scroll through an immensely large list of pages, posts, events, etc. It would be nice to separate that like this: 
```js
LinkFeature({
  // this only adds new fields:
  fields: [
    {
      type: 'relationship',
      relationTo: ['posts', 'events'],
      name: 'dynamic',
      label: 'Internal link to dynamic content',
    },
    {
      type: 'relationship',
      relationTo: ['images', 'documents'],
      name: 'assets',
      label: 'Internal link to images or documents'
    },
  ],
})
```
Furthermore we could then use `filterOptions` on the relationship field which would allow great flexibility.
My problem is that I'm not familiar with React and I can't write a new linkFeature. I would like to modify the existing one but I don't know how to modify the existing fields.

**Comment by BrianJM** [comment]
Did you find a solution?

**Reply by unonweb** [reply]
No, sorry.

**Comment by zakinadhif** [comment]
I'm having the same problem, additionally the serialized rich-text only store document ID, but sometimes you want documents to be referred using its slug, in that case you'd need to inquire what slug does the document with this ID has, which hurts performance if there's a lot of links.

**Comment by zakinadhif** [comment]
I suppose you can replace the `LinkFeature` serializer to serialize the slug instead of the ID of the document, but that's kinda complicated.

**Comment by damnitrahul** [comment]
Facing the same issue here. I should ideally be able to filter the data to reduce the size of the response.
